engine/src/io/data_provider/sql/AbstractSQLDataProvider.cpp

```diff
@@ -53,11 +151,20 @@ std::string abstractsql_data_provider::build_select_from() const {
       }
     }
   }
-  return "SELECT " + cols + "FROM " + this->sql.table;
-}
 
-std::string abstractsql_data_provider::build_limit_offset(size_t offset) const {
-  return " LIMIT " + std::to_string(this->sql.table_batch_size) + " OFFSET " + std::to_string(offset);
+  const size_t offset = this->sql.table_batch_size * (this->total_number_of_nodes * batch_index + this->self_node_idx);
+  std::string limit = " LIMIT " + std::to_string(this->sql.table_batch_size) + " OFFSET " + std::to_string(offset);
+  auto ret = "SELECT " + cols + "FROM " + this->sql.table;
+
+  if (sql.table_filter.empty()) {
```

Comment:
if your table_filter is not empty and the `where` is not empty you can do:
`" where (" + sql.table_filter + ") AND (" + this->where + ")"`

Reply:
Good idea!

Reply:
Done ty